script/generateTypeMetadata.js

```diff
@@ -0,0 +1,37 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const readline = require("readline")
+
+// eslint-disable-next-line id-length
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+    terminal: false,
+})
+
+const fieldToMetadata = (field, required) =>
+    `${field}: {required: ${required.includes(field)}}`
+
+const schemaToString = (type, schema) => {
+    const fields = Object.keys(schema.properties)
+    const { required } = schema
+    const fieldMetadata = `export const ${type.toLowerCase()}FieldMetadata: Record<string, FieldMetadata> = {${fields
```

Comment:
Instead of defining this const `fieldMetadata`, we  could just return the right hand side of the assignment.